### ISSUE TITLE
test

### ISSUE THREAD

**Issue by Mairafatoretto** [histocartography/zoommil#3]
In the train.py we are using all features to train, validate and test. However the model did not to extract features from all patches for the test. My question is if in zoommil/trainer.py only the top-k features are being considered for the test or all patches are being used to make the inference?

**Comment by kevthan** [comment]
Hi, as seen here:

- https://github.com/histocartography/zoommil/blob/main/zoommil/models/zoommil.py#L107
- https://github.com/histocartography/zoommil/blob/main/zoommil/models/zoommil.py#L139

only the top K features with highest attention are used for inference at test time.

**Comment by Mairafatoretto** [comment]
Thank you for the ansewer Kevin. But i did not to find in the code, where did you extract the test features for only top k patches.

And, as default the k are igual to 12. Why?

**Comment by kevthan** [comment]
You are right, the lines I linked implement the top-k selection but not the on-the-fly feature extraction from patches. We have not included this part in the code yet. I will try to add it when I get some time in the future.

K is a hyperparameter. The default is 12 since that is the number we used for the BRIGHT dataset.
